### PR TITLE
Add ZooKeeper pack with non_blocking_lease action

### PR DESCRIPTION
**Pull request by codyaray**
### Status 

- new pack, minimally complete

### Description

Pack which allows integration with Apache ZooKeeper.

### Checklist (tick everything that applies)

- [x] Metadata: pack.yaml, icon, structure (required for new packs)
- [ ] Version bump (required for changed packs)
- [ ] Code linting (required, can be done after the PR checks)
- [ ] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)

### CODE REVIEW

**Comment by Kami** [comment]
Nice, thanks. :+1: 

Will review it as soon as possible.

File: packs/zookeeper/actions/non_blocking_lease.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from datetime import timedelta
+from kazoo.client import KazooClient
+from socket import gethostname
+from st2actions.runners.pythonrunner import Action
+import sys
```
Comment (lakshmi-kannan) [inline packs/zookeeper/actions/non_blocking_lease.py]: Nit: Heh. Please order imports and group them based on PEP guidelines.

File: packs/zookeeper/config.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (lakshmi-kannan) [inline packs/zookeeper/config.yaml]: Please switch to using new config.schema approach.

File: packs/zookeeper/config.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+zookeeper_hosts: "localhost:2181"
+zookeeper_root: "/"
```
Comment (lakshmi-kannan) [inline packs/zookeeper/config.yaml]: Wondering why you decided to put this in config as opposed to action param?

**Comment by lakshmi-kannan** [comment]
Nice!

File: packs/zookeeper/config.yaml (hunk 2 above)
Comment (codyaray) [inline packs/zookeeper/config.yaml]: This was a PR before the new config.schema approach was released. ;)

File: packs/zookeeper/config.yaml (hunk 3 above)
Comment (codyaray) [inline packs/zookeeper/config.yaml]: We could make it overridable in action params, but this was the simplest thing that solved the immediate need. My basic reasoning is that StackStorm users don't (usually) care about URLs, so specify them system-wide. Of course, that assumption fails when you have multiple Zookeeper clusters that you want to integrate with; but we can cross that bridge when we come to it. :)

Comment (lakshmi-kannan) [inline packs/zookeeper/config.yaml]: Yeah, I'd actually prefer to namespace it like /stackstorm/leases/ but I think anyone familiar with ZK can configure it the way they want.

File: packs/zookeeper/config.yaml (hunk 2 above)
Comment (lakshmi-kannan) [inline packs/zookeeper/config.yaml]: I figured. It's a suggestion :P.

File: packs/zookeeper/actions/non_blocking_lease.py (hunk 1 above)
Comment (codyaray) [inline packs/zookeeper/actions/non_blocking_lease.py]: Fixed. Assuming you want alphanumeric ordering within groups, which PEP-8 doesn't actually specify. https://www.python.org/dev/peps/pep-0008/#imports :)

Comment (lakshmi-kannan) [inline packs/zookeeper/actions/non_blocking_lease.py]: yeah, the latest commit looks good.

File: packs/zookeeper/config.yaml (hunk 3 above)
Comment (codyaray) [inline packs/zookeeper/config.yaml]: Yeah, I set it to "/stormcron" in my config too. I could see needing to override the root path on an action-by-action basis before needing to override the host. But that can be done as a follow-up. :)

**Pull request merged**